internal/api/mfa.go

```diff
@@ -245,7 +299,33 @@ func (a *API) VerifyFactor(w http.ResponseWriter, r *http.Request) error {
 		return badRequestError("%v has expired, verify against another challenge or create a new challenge.", challenge.ID)
 	}
 
-	if valid := totp.Validate(params.Code, factor.Secret); !valid {
+	valid := totp.Validate(params.Code, factor.Secret)
+	if config.Hook.MFAVerificationAttempt.Enabled {
+		input := hooks.MFAVerificationAttemptInput{
+			UserID:   user.ID,
+			FactorID: factor.ID,
+			Valid:    valid,
+		}
+		output := &hooks.MFAVerificationAttemptOutput{}
+
+		if err := a.invokeHook(ctx, input, output); err != nil {
+			return errors.New(err.Error())
+		}
+		if terr := models.NewAuditLogEntry(r, a.db, user, models.InvokeAuthHookAction, r.RemoteAddr, map[string]interface{}{
+			"extensibility_point_event": hooks.MFAVerificationAttempt,
+			"factor_id":                 factor.ID,
+			"URI":                       config.Hook.MFAVerificationAttempt.URI,
```

Comment:
Let's not audit log this at all at this stage. I don't think this is worthy of an "audit" event.